src/stronghold/mod.rs

```diff
@@ -280,6 +280,11 @@ impl StrongholdAdapter {
         if let Some(mut key) = self.key.lock().await.take() {
             key.zeroize();
         }
+
+        // Unload Stronghold, but we can't do much about the errors.
+        if let Err(err) = self.unload_stronghold_snapshot().await {
```

Comment:
This needs to be done before clearing the key, otherwise it will always fail

Reply:
Oops, yes.